src/utils/utils.js

```diff
@@ -67,15 +67,213 @@ export const getConfigObjectsForAnalytics = (configurations, groupCode) => {
     })
 
     // Create an object to index datasets by their IDs
-    const indexedDatasets = {};
+    const indexedDatasets = {}
 
     datasets.forEach((dataset) => {
-        indexedDatasets[dataset.id] = dataset;
-    });
+        indexedDatasets[dataset.id] = dataset
+    })
 
     const configsObj = {
         dataElementsAndIndicators: indexedNumerators,
         dataSets: indexedDatasets,
     }
     return configsObj
 }
+
+// gets a list of retions in which the reporting rate score was lower than the threshold
+const getRegionsWithLowScore = (filterd_datasets, key) => {
+    const dataset = filterd_datasets[key]
+    if (!dataset) {
+        return [] // Return an empty array if the key is not found
+    }
+
+    const regionsWithLowScore = dataset
+        .filter((entry) => parseFloat(entry.score) < entry.threshold)
+        .map((entry) => entry.orgUnitLevelsOrGroups)
+
+    return regionsWithLowScore
+}
+
+// returns a JSON formatted object from the table-like format from analytics
+const getJsonObjectsFormatFromTableFormat = ({
+    headers,
+    rows,
+    metaData,
+    mappedConfigurations,
+    calculatingFor,
+}) => {
+    const restructuredData = {}
+
+    for (const row of rows) {
+        const rowData = {}
+        const ouHeaderIndex = headers.map((header) => header.name).indexOf('ou')
+        const dsNameIndex = headers.map((header) => header.name).indexOf('dx')
+
+        for (let i = 0; i < headers.length; i++) {
+            const header = headers[i]
+            const columnName = header.name
+            const columnValue = row[i]
+            // Add the key-value pair to the row data object
+            rowData[columnName] = columnValue
+        }
+
+        //
+        rowData['orgUnitLevelsOrGroups'] =
+            metaData.items[row[ouHeaderIndex]].name
+        rowData['dataset_name'] =
+            metaData.items[row[dsNameIndex]].name +
+            (calculatingFor == 'completeness' ? '' : ' on time')
+        const currentDataSetId = row[0].split('.')[0]
+        if (calculatingFor == 'completeness') {
```

Comment:
it'd be better to generally use strict equalities `calculatingFor === 'completeness'`, though in this case it doesn't make a difference